# 🚢 Titanic Survival Analysis – Descriptive Statistics

This notebook contains the step-by-step process of cleaning, analyzing, and drawing insights from the Titanic dataset using Python (Pandas).
The objective is to explore how features like gender, age, passenger class, and fare impacted survival rates.

The results of this analysis were later visualized using Power BI and compiled into a professional report.

In [2]:
import pandas as pd

In [2]:
df = pd.read_excel(r'Descriptive Project.xlsx', sheet_name= 'working_directory') #workbook has two files one is for backup and other is for work

In [3]:
df

PassengerId  Survived Survived.1  Pclass   Class  \
0              1         0   Perished       3   Third   
1              2         1   Survived       1   First   
2              3         1   Survived       3   Third   
3              4         1   Survived       1   First   
4              5         0   Perished       3   Third   
..           ...       ...        ...     ...     ...   
886          887         0   Perished       2  Second   
887          888         1   Survived       1   First   
888          889         0   Perished       3   Third   
889          890         1   Survived       1   First   
890          891         0   Perished       3   Third   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 14 columns]

In [4]:
df = df.drop(columns =['PassengerId', 'Survived.1', 'Ticket', 'Name', 'Cabin', 'Survived.1', 'Class']) #Deleting unwanted columns

In [5]:
df

Survived  Pclass     Sex   Age  SibSp  Parch     Fare Embarked
0           0       3    male  22.0      1      0   7.2500        S
1           1       1  female  38.0      1      0  71.2833        C
2           1       3  female  26.0      0      0   7.9250        S
3           1       1  female  35.0      1      0  53.1000        S
4           0       3    male  35.0      0      0   8.0500        S
..        ...     ...     ...   ...    ...    ...      ...      ...
886         0       2    male  27.0      0      0  13.0000        S
887         1       1  female  19.0      0      0  30.0000        S
888         0       3  female   NaN      1      2  23.4500        S
889         1       1    male  26.0      0      0  30.0000        C
890         0       3    male  32.0      0      0   7.7500        Q

[891 rows x 8 columns]

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 8 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Survived  891 non-null    int64  
 1   Pclass    891 non-null    int64  
 2   Sex       891 non-null    object 
 3   Age       714 non-null    float64
 4   SibSp     891 non-null    int64  
 5   Parch     891 non-null    int64  
 6   Fare      891 non-null    float64
 7   Embarked  889 non-null    object 
dtypes: float64(2), int64(4), object(2)
memory usage: 55.8+ KB


### Data Quality Check – Initial Observations
- `Age` column has **177 missing values** (714/891)
- `Embarked` column has **2 missing values**
- All other columns are complete

In [7]:
df.describe()

Survived      Pclass         Age       SibSp       Parch        Fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200

Following the rule that if **standard deviation > 50% of the mean**, it signals a **possible presence of outliers**.
Based on `df.describe()`, the above columns show signs of potential outliers.


In [8]:
df.nunique() #it shows the unique values contained by the dataset

Survived      2
Pclass        3
Sex           2
Age          88
SibSp         7
Parch         7
Fare        248
Embarked      3
dtype: int64

The **agenda of using the above code is to quick scan the dataset** to spot the errors.
Focused columns are   `Embarked` and `Sex` but everything seems okay here.

In [9]:
df['Sex'].unique() #manually checking for errors in the Sex column

array(['male', 'female'], dtype=object)

In [10]:
df['Embarked'].unique() #manually checking for errors in the Embarked column

array(['S', 'C', 'Q', nan], dtype=object)

### Checked Unique Values
- Used `df.nunique()` to get a quick count of unique values in each column.
- Then used `df['Sex'].unique()` and `df['Embarked'].unique()` to manually inspect actual values in key categorical columns.
- Found that `Embarked` has a 4th value `'nan'` (missing), which is not shown in `nunique()` since it excludes nulls by default.
- This confirms the presence of missing values and highlights why **manual checks are important** after a quick scan.


## Handling Missing Values using imputation

### Handled – Age (BELOW)
- `Age` had ~20% missing values (177 rows).
- Instead of dropping, we chose **median imputation** to preserve data.
- Reason: Median is less sensitive to outliers and better suited for skewed numerical data.


In [11]:
df['Age'] = df['Age'].fillna(df['Age'].median()) #filling the age column where the value = nan

In [12]:
df['Age'].isnull().sum() #checking the number of nulls in the age column

0

### Handled Values – Embarked (BELOW)
- `Embarked` had only 2 missing values.
- Used **mode imputation**, as it's a categorical column.
- Mode fills in the most frequent category, minimizing bias without data loss.


In [13]:
df['Embarked'] = df['Embarked'].fillna(df['Embarked'].mode()[0])

In [14]:
df['Embarked'].isnull().sum() #checking the number of nulls in the embarked column

0

## Frequency Distribution

In [15]:
df['Sex'].value_counts(normalize=True).round(2)

Sex
male      0.65
female    0.35
Name: proportion, dtype: float64

`Male` proportion is **65%** and `Women` proportion is **35%**.

In [16]:
df['Pclass'].value_counts(normalize=True).round(2)

Pclass
3    0.55
1    0.24
2    0.21
Name: proportion, dtype: float64

Passenger Class `1`, `2` and `3` has proportion **24%**, **21%** and **55%** respectively.

In [17]:
df['Embarked'].value_counts(normalize=True).round(2)

Embarked
S    0.73
C    0.19
Q    0.09
Name: proportion, dtype: float64

Embarked `S`, `C` and `Q` has proportion **73%**, **19%** and **9%** respectively.

In [18]:
df['Survived'].value_counts(normalize=True).round(2)

Survived
0    0.62
1    0.38
Name: proportion, dtype: float64

`Death` proportion is **62%** and `Survived` proportion is **38%**.

### Summary of Categorical Insights
- The dataset is **male-dominant**, with 65% male passengers.
- Most passengers were in **3rd class (55%)**, indicating a lower socio-economic group.
- Majority of passengers **embarked from port S (73%)**, making it the primary departure location.
- The dataset shows an **imbalanced survival rate**, with 62% deaths and 38% survival.


## Numerical Column Analysis

In [19]:
df.describe().round(2)

Survived  Pclass     Age   SibSp   Parch    Fare
count    891.00  891.00  891.00  891.00  891.00  891.00
mean       0.38    2.31   29.36    0.52    0.38   32.20
std        0.49    0.84   13.02    1.10    0.81   49.69
min        0.00    1.00    0.42    0.00    0.00    0.00
25%        0.00    2.00   22.00    0.00    0.00    7.91
50%        0.00    3.00   28.00    0.00    0.00   14.45
75%        1.00    3.00   35.00    1.00    0.00   31.00
max        1.00    3.00   80.00    8.00    6.00  512.33

`Survived` and `Pclass` are **Binary categorical** and **Categorical Ordinal** data. So, we need to specify which column we want to use to see the `Measure of Dispersion`.

In [20]:
df[['Age', 'SibSp', 'Parch', 'Fare']].describe().round(2)

Age   SibSp   Parch    Fare
count  891.00  891.00  891.00  891.00
mean    29.36    0.52    0.38   32.20
std     13.02    1.10    0.81   49.69
min      0.42    0.00    0.00    0.00
25%     22.00    0.00    0.00    7.91
50%     28.00    0.00    0.00   14.45
75%     35.00    1.00    0.00   31.00
max     80.00    8.00    6.00  512.33

Except `Age` column all the other three columns showing the chances of potential **outliers**.

### Outlier Detection Using IQR
To detect outliers in numerical columns, the Interquartile Range (IQR) is going to be used here.
- **Q1** (25th percentile) and **Q3** (75th percentile) were calculated.
- **IQR** = Q3 - Q1
- **Lower Bound** = Q1 - 1.5 × IQR
- **Upper Bound** = Q3 + 1.5 × IQR

Values outside these bounds are considered potential outliers.


In [21]:
def iqr_summary(df, columns): #defined function to maintain code readability and usability for future use
    records = []

    for col in columns:
        Q1 = df[col].quantile(0.25)
        Q3 = df[col].quantile(0.75)
        IQR = Q3 - Q1
        lower = Q1 - 1.5 * IQR
        upper = Q3 + 1.5 * IQR
        records.append({
            'Column': col,
            'Q1': round(Q1, 2),
            'Q3': round(Q3, 2),
            'IQR': round(IQR, 2),
            'Lower Bound': round(lower, 2),
            'Upper Bound': round(upper, 2)
        })

    return pd.DataFrame(records)


In [22]:
iqr_summary(df, ['Age', 'SibSp', 'Parch', 'Fare'])

Column     Q1    Q3    IQR  Lower Bound  Upper Bound
0    Age  22.00  35.0  13.00         2.50        54.50
1  SibSp   0.00   1.0   1.00        -1.50         2.50
2  Parch   0.00   0.0   0.00         0.00         0.00
3   Fare   7.91  31.0  23.09       -26.72        65.63

Now, that we have IQR and Bounds calculated. It's time to - Detect **actual outliers** using those bounds

In [23]:
outliers_age = df[(df['Age'] < 2.5) | (df['Age'] > 54.5)]
outliers_sibsp = df[(df['SibSp'] < -1.5) | (df['SibSp'] > 2.5)]
outliers_fare = df[(df['Fare'] < -26.72) | (df['Fare'] > 65.63)]

### Note on 'Parch' Column
The `Parch` column was excluded from outlier detection using the IQR method because:
- Both Q1 and Q3 were 0, resulting in an IQR of 0.
- This indicates that at least 75% of the values are zero.
- With no meaningful spread, the IQR method is not suitable for identifying outliers in this column.

In [24]:
outlier_counts = pd.DataFrame({
    'Column': ['Age', 'SibSp', 'Fare'],
    'Outlier Count': [
        len(outliers_age),
        len(outliers_sibsp),
        len(outliers_fare)
    ]
})
outlier_counts

Column  Outlier Count
0    Age             66
1  SibSp             46
2   Fare            116

### Outlier Summary
- `Fare` has the highest number of outliers, suggesting a wide range in ticket prices.
- `Age` contains some high or low extremes, possibly indicating very young or old passengers.
- `SibSp` has few outliers, mostly due to large families or solo travelers.
- These outliers will be noted but **not removed**, as they may carry meaningful patterns.


## Central Tendency & Spread Analysis

In [25]:
df[['Age', 'SibSp', 'Fare']].agg(['mean', 'median', 'std', 'min', 'max']).round(2)

Age  SibSp    Fare
mean    29.36   0.52   32.20
median  28.00   0.00   14.45
std     13.02   1.10   49.69
min      0.42   0.00    0.00
max     80.00   8.00  512.33

### Numerical Summary – Central Tendency & Spread

- **Age**: Stable distribution with mean (29.36) close to median (28.0); standard deviation (13.02) is moderate. Indicates a symmetric, well-distributed variable.
- **SibSp**: Right-skewed. Mean (0.52) is higher than median (0.0), and standard deviation (1.1) suggests variability due to large family groups.
- **Fare**: Highly skewed. Mean (32.2) is more than double the median (14.45), and a large standard deviation (49.69) indicates strong outlier influence and a wide spread in ticket prices.


## Relationship Comparison

In [26]:
df.groupby('Sex')['Survived'].mean().round(2)

Sex
female    0.74
male      0.19
Name: Survived, dtype: float64

### Survival Rate by Gender
- Female passengers had a significantly higher survival rate (**74%**) compared to males (**19%**).
- This reflects the "women and children first" policy followed during the Titanic evacuation.


> Note: The 74% (females) and 19% (males) survival rates show how many from each group survived.
> They don’t add up to 100% because they are **separate group-wise percentages**, not parts of a whole.
> No data is missing — it’s just how the calculation works.


## Pclass Vs Survived

In [27]:
df.groupby('Pclass')['Survived'].mean().round(2)

Pclass
1    0.63
2    0.47
3    0.24
Name: Survived, dtype: float64

### Survival Rate by Passenger Class
- 1st Class: **63%** survival rate
- 2nd Class: **47%** survival rate
- 3rd Class: **24%** survival rate

This shows that passengers in higher classes had a significantly better chance of survival, possibly due to better cabin locations or priority during evacuation.


## Embarked Vs Survived

In [28]:
df.groupby('Embarked')['Survived'].mean().round(2)

Embarked
C    0.55
Q    0.39
S    0.34
Name: Survived, dtype: float64

### Survival Rate by Embarkation Port
- Cherbourg (`C`): **55%** survival rate
- Queenstown (`Q`): **39%** survival rate
- Southampton (`S`): **34%** survival rate

Passengers who boarded at Cherbourg had the highest survival rate. This may reflect differences in passenger class or ship sections associated with each port.


## Age Vs Survived

In [29]:

df.groupby('Survived')['Age'].mean().round(2)

Survived
0    30.03
1    28.29
Name: Age, dtype: float64

### Survival Rate by Age (Average Comparison)
- The average age of those who did not survive was **30.03 years**.
- The average age of survivors was **28.29 years**.
- The age difference is small, suggesting age alone did not strongly influence survival.


In [33]:
with pd.ExcelWriter(r"C:\excel_practice\Portfolio\Work_to_show\Descriptive\Descriptive Project.xlsx", mode='a', engine='openpyxl', if_sheet_exists='replace') as writer:
    df.to_excel(writer, sheet_name='working_directory', index=False)

---

## ✅ Summary

The dataset was successfully cleaned and analyzed using descriptive statistical methods.
Key features were explored using measures like mean, median, standard deviation, and IQR.
Insights drawn from the analysis formed the foundation of the Power BI dashboard and final report.

📌 This notebook serves as the backend of the project — tracking every step taken during analysis.

---


# Descriptive Analysis Summary – Titanic Dataset

## Objective
The goal of this project was to apply **descriptive analysis techniques** to the Titanic dataset to extract meaningful insights from both categorical and numerical data. The focus was on **cleaning, summarizing, and understanding** the structure of the data — laying a foundation for further analysis or visualization.

---

## Data Preparation & Cleaning
- Unnecessary columns (`PassengerId`, `Name`, `Ticket`, `Cabin`, etc.) were removed to simplify analysis.
- Missing values were handled:
  - `Age`: Imputed using **median** due to slight skewness and outliers.
  - `Embarked`: Imputed using **mode** since it’s categorical and had only 2 missing entries.
- `Pclass` and `Survived` were kept as categorical despite being numeric.

---

## Categorical Feature Insights
- **Sex**: 65% male, 35% female. Females had a **74% survival rate** vs. 19% for males.
- **Pclass**: Most passengers were in 3rd class (55%), which had the **lowest survival rate** at 24%.
- **Embarked**: Majority embarked from `S` (73%). Those who boarded at `C` had the **highest survival rate** (55%).
- **Survived**: 38% survived, 62% perished — showing a significant imbalance.

---

## Numerical Feature Insights
- **Age**: Mean = 29.36, Median = 28.0 → fairly stable distribution.
- **Fare**: Mean = 32.2, Median = 14.45 → **highly skewed**, affected by outliers.
- **SibSp** and **Parch**: Right-skewed, mostly zero. Parch had no spread, so it was excluded from outlier analysis.

### Outlier Detection (IQR Method)
- Outliers were detected in `Fare`, `SibSp`, and `Age`, with **Fare** showing the highest deviation.
- No outliers were removed; they were considered potentially meaningful.

---

## Feature Relationships
- **Sex vs Survived**: Females survived significantly more than males (74% vs 19%).
- **Pclass vs Survived**: 1st class had 63% survival; 3rd class had only 24%.
- **Embarked vs Survived**: Passengers from Cherbourg (`C`) had the highest survival rate.
- **Age vs Survived**: Little difference in average age of survivors vs non-survivors (28.29 vs 30.03).

---

## Key Takeaways
- Survival strongly correlates with **gender and class**.
- **Ticket fare** and **embarkation point** show indirect influence, possibly through class.
- The dataset is **cleaned, imputed, and structured** — ready for dashboarding or exploratory data analysis (EDA).

---

## Next Step: Visualization in Power BI
This dataset is now ready for visual exploration in Power BI. The insights found here will guide the story and help structure the visuals in a meaningful way.
